compiler/hsSyn/HsUtils.hs

```diff
@@ -359,14 +359,20 @@ mkHsStringPrimLit fs
   = HsStringPrim noSourceText (fastStringToByteString fs)
 
 -------------
+-- KWF: These don't seem to be used anywhere in the compiler, and because we
```

Comment:
How difficult do you think it would be to make such a tool? A first pass at it might not even need to use the GHC API; there's only a very limited amount of parsing required to find use-sites, it seems: exclude mentions in module export list, exclude the selfsame definition site, and otherwise note uses. This would generate false-negatives (things which *could* be deleted but aren't reported) in the case of name-shadowing or even the same unqualified name being used in two different modules, but it would likely be a useful start.

Reply:
That sounds dirty, to be honest. I'm sure you'd find other strange behavior. But in any case, there are other fish to fry.